### PR TITLE
Add GH Actions CI #41

### LINKED ISSUE #41: Add CI for PRs

### PR DESCRIPTION
**Pull request by kabir**
Fixes #41

Follow-up issues to reenable the tests from the 'Skip some tests which are hanging locally' commit:
https://github.com/wise-agents/wise-agents/issues/163
https://github.com/wise-agents/wise-agents/issues/164
https://github.com/wise-agents/wise-agents/issues/165

This PR introduces GH Actions CI, so they won't get run until this PR is merged. https://github.com/kabir/wise-agents/actions/runs/10406094994 contains the latest run of the code contained in this branch.

### CODE REVIEW

File: tests/wiseagents/test_tools.py [hunk 1]
```diff
@@ -83,6 +84,7 @@ def stop(self):
 #Install ollam from their official website https://ollama.com/
 #run ollama with the following command: ollama run llama3.1
 
+@pytest.mark.skip(reason="does not pass CI/CD")
```
Comment (fjuma) [inline tests/wiseagents/test_tools.py:87]: This tools test and test_coordinator both make use of llama 3.1 so we'll also need to set up Ollama for these tests to pass.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/pyproject.toml') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+    - name:  Update pip
+      run: |
+        python -m pip install --upgrade pip
+    # The LMM takes too much space, so we will disable the steps to install it, and not run those tests
+    - name: Install HuggingFaces Hub
+      # Skip LMM
+      if: env.ENABLE_LMM != 'true'
```
Comment (fjuma) [inline .github/workflows/build-and-test.yml:67]: s/LMM/LLM

Comment (fjuma) [reply .github/workflows/build-and-test.yml:67]: If you're not around, no worries, we can merge as is and then fix after.

File: tests/wiseagents/test_yaml_deserializer.py [hunk 3]
```diff
@@ -1,10 +1,13 @@
+import pytest
+
 from wiseagents import WiseAgent
 from wiseagents.llm import WiseAgentRemoteLLM
 from wiseagents.graphdb import LangChainWiseAgentGraphDB, Neo4jLangChainWiseAgentGraphDB
 import yaml
 import logging, pathlib
 
 
+@pytest.mark.skip(reason="does not pass CI/CD")
```
Comment (fjuma) [inline tests/wiseagents/test_yaml_deserializer.py:10]: The tests in this file call `llm.process_single_prompt` so I think we can mark them as `@pytest.mark.needsllm`

File: tests/wiseagents/test_tools.py (hunk 1 above)
Comment (kabir) [reply tests/wiseagents/test_tools.py:87]: Ah, I missed that since I didn't see a script for this. Will fix next week

**Review by fjuma — APPROVED** [review]
Thanks @kabir!

**Pull request merged**